static/js/unread_ops.js

```diff
@@ -73,6 +74,53 @@ export function process_read_messages_event(message_ids) {
     unread_ui.update_unread_counts();
 }
 
+export function process_unread_messages_event({message_ids, message_details}) {
+    // This is the reverse of  process_unread_messages_event.
+    message_ids = unread.get_read_message_ids(message_ids);
+    if (message_ids.length === 0) {
+        return;
+    }
+
+    for (const message_id of message_ids) {
+        const message = message_store.get(message_id);
+
+        if (message) {
+            message.unread = true;
+        }
+
+        const message_info = message_details[message_id];
+
+        let user_ids_string;
+
+        if (message_info.type === "private") {
+            user_ids_string = message_info.sender_id.toString();
+        } else if (message_info.type === "huddle") {
+            user_ids_string = people.pm_lookup_key(message_info.sender_id_str);
```

Comment:
I think this should be `event_details.user_ids_string`, since we renamed that.  And we might want to add pm/huddle test coverage to catch things like this.